### DISCUSSION TITLE
Do enhance maps work if I customize a resolver's args types?

### DISCUSSION THREAD

**Discussion by omar-dulaimi** [MichalLytek/typegraphql-prisma#74] [Q&A] [answered]
Hello,

So basically I have taken out some of the generated resolvers, changed their names, some changed their args types to expose just a small number of fields and hid the rest.

I tried to apply enhance maps but they didn't work. Is that intentional?

**Comment by MichalLytek** [comment, marked as answer]
Yes, it works only with the original generated artifacts because the generator have no idea about your changed names or args.

You can always call the decorators manually, like `IsString()(MyResolver.prototype, "myMethod");`.
But as you've changed the names and types, you can just put the decorators by yourself 😉
